File: Software/Backend/BusinessLogicLayer/AppLogic/Users/UpdateUserProfile/UpdateUserRequest.cs
using BusinessLogicLayer.AppLogic.Skills;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessLogicLayer.AppLogic.Users.UpdateUserProfile
{
    public record UpdateUserRequest
    {
        public int UserId { get; set; }
        public string? Name { get; set; }
        public string? Address { get; set; }
        public string? Phone { get; set; }
        public string? ProfilePicture { get; set; }
        public List<int>? AddSkills { get; set; } = new List<int>();
        public List<int>? RemoveSkills { get; set; } = new List<int>();
    }
}

File: Software/Frontend/ws/src/main/java/hr/foi/air/baufind/ws/request/UpdateUserRequest.kt
package hr.foi.air.baufind.ws.request

data class UpdateUserRequest(
    val userId : Int,
    val name: String? = null,
    val address: String? = null,
    val phone: String? = null,
    val profilePicture: String? = null,
    val addSkills: List<Int> = emptyList(),
    val removeSkills: List<Int> = emptyList()
)
